kDriveCore/Data/UploadQueue/Queue/UploadQueue+Queue.swift

```diff
@@ -351,20 +351,24 @@ extension UploadQueue: UploadQueueable {
             Log.uploadQueue("suspend queue")
             self.suspendAllOperations()
 
-            var uploadingFilesIds = [String]()
-            try? self.transactionWithUploadRealm { realm in
-                let uploadingFiles = self.getUploadingFiles(withParent: parentId,
-                                                            userId: userId,
-                                                            driveId: driveId,
-                                                            using: realm)
-                Log.uploadQueue("cancelAllOperations count:\(uploadingFiles.count) parentId:\(parentId)")
-                uploadingFilesIds = uploadingFiles.map(\.id)
-                Log.uploadQueue("cancelAllOperations IDS count:\(uploadingFilesIds.count) parentId:\(parentId)")
+            let uploadingFiles = self.getUploadingFiles(withParent: parentId,
+                                                        userId: userId,
+                                                        driveId: driveId)
+
+            let uploadingFilesIds = uploadingFiles.map(\.id)
+            Log.uploadQueue("cancelAllOperations count:\(uploadingFiles.count) parentId:\(parentId)")
+            Log.uploadQueue("cancelAllOperations IDS count:\(uploadingFilesIds.count) parentId:\(parentId)")
 
+            try? self.transactionWithUploadRealm { realm in
                 // Delete all the linked UploadFiles from Realm. This is fast.
                 try? realm.safeWrite {
                     Log.uploadQueue("delete all matching files count:\(uploadingFiles.count) parentId:\(parentId)")
-                    realm.delete(uploadingFiles)
+                    for fileId in uploadingFilesIds {
```

Comment:
This does extra work compared to the old implementation, need to think of a better way to delete objects.

Reply:
Reworked with a IN predicate